keps/sig-cluster-lifecycle/kubeadm/20190424-kubeadm-for-windows.md

```diff
@@ -171,58 +183,37 @@ Windows related adjustments to default paths might be required.
 
 #### Windows vs Linux host paths
 
-Kubeadm makes a number of non-portable assumptions about paths. E.g. “/etc/kubernetes” is a hardcoded path in kubeadm.
+Kubeadm makes use of several Linux-specific paths. E.g. “/etc/kubernetes” is a hardcoded path in kubeadm. We intend to use these paths on Windows as well, though we would be open to making them follow Windows path standards later.
 
-We will use "C:\kubernetes" to hold the paths that are normally created for Linux.
```

Comment:
this was a nice detail and it think maybe we should continue using a single location for all assets and binaries?

Reply:
This wasn't true, at least not in the Powershell implementation in sig-windows-tools as it stands right now. There's still files that get placed by kubeadm into `/etc` and `/var`. If we're just talking about the Kubernetes binaries, I am more on board with it, but changing the location of the bootstrap kubeconfig, certs, and kubelet config are going to be more difficult.

Reply:
So I guess if we want to do this still in the KEP we should be explicit about what it means, right now it seems to mean "every file that's created will be rooted in `C:\kubernetes`"

Reply:
i think that the KEP should just vaguely mention that we want to lock all important files and locations, but not specify how (i.e. we have some options and we are not sure how.). it can mention that downloaded artifacts will go in `c:\kubernetes`. but i think the script must allow a custom drive letter.

i think this is worth investigating:
https://github.com/kubernetes/enhancements/pull/1456/files/3723b4e9f324deea412e5db1087da170185abe38#r367146431